### DISCUSSION TITLE
Airflow showing 504 Gateway erorr message

### DISCUSSION THREAD

**Discussion by seyal84** [apache/airflow#22973] [General]
Azure AKS 

### Apache Airflow version

2.2.1

### What happened

Python version: 3.7.12 Airflow version: 2.2.1 Node: airflow-webserver-5578579dc8-6zj2t ------------------------------------------------------------------------------- Traceback (most recent call last):   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/connection.py", line 175, in _new_conn     (self._dns_host, self.port), self.timeout, **extra_kw   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/util/connection.py", line 73, in create_connection     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):   File "/usr/local/lib/python3.7/socket.py", line 752, in getaddrinfo     for res in _socket.getaddrinfo(host, port, family, type, proto, flags): socket.gaierror: [Errno -3] Temporary failure in name resolution During handling of the above exception, another exception occurred: Traceback (most recent call last):   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen     chunked=chunked,   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 382, in _make_request     self._validate_conn(conn)   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn     conn.connect()   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/connection.py", line 358, in connect     conn = self._new_conn()   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/connection.py", line 187, in _new_conn     self, "Failed to establish a new connection: %s" % e urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f7d5ed74490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution During handling of the above exception, another exception occurred: Traceback (most recent call last):   File "/home/airflow/.local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send     timeout=timeout   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]   File "/home/airflow/.local/lib/python3.7/site-packages/urllib3/util/retry.py", line 574, in increment     raise MaxRetryError(_pool, url, error or ResponseError(cause)) urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /5989ece0-f90e-40bf-9c79-1a7beccdb861/oauth2/v2.0/token (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f7d5ed74490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')) During handling of the above exception, another exception occurred: Traceback (most recent call last):   File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app     response = self.full_dispatch_request()   File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request     rv = self.handle_user_exception(e)   File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception     reraise(exc_type, exc_value, tb)   File "/home/airflow/.local/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise     raise value   File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request     rv = self.dispatch_request()   File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request     return self.view_functions[rule.endpoint](**req.view_args)   File "/home/airflow/.local/lib/python3.7/site-packages/flask_appbuilder/security/views.py", line 655, in oauth_authorized     resp = self.appbuilder.sm.oauth_remotes[provider].authorize_access_token()   File "/home/airflow/.local/lib/python3.7/site-packages/authlib/integrations/flask_client/remote_app.py", line 76, in authorize_access_token     token = self.fetch_access_token(**params)   File "/home/airflow/.local/lib/python3.7/site-packages/authlib/integrations/base_client/remote_app.py", line 110, in fetch_access_token     token = client.fetch_token(token_endpoint, **kwargs)   File "/home/airflow/.local/lib/python3.7/site-packages/authlib/oauth2/client.py", line 205, in fetch_token     headers=headers, **session_kwargs   File "/home/airflow/.local/lib/python3.7/site-packages/authlib/oauth2/client.py", line 221, in _fetch_token     method, url, data=body, headers=headers, auth=auth, **kwargs)   File "/home/airflow/.local/lib/python3.7/site-packages/authlib/integrations/requests_client/oauth2_session.py", line 113, in request     method, url, auth=auth, **kwargs)   File "/home/airflow/.local/lib/python3.7/site-packages/requests/sessions.py", line 542, in request     resp = self.send(prep, **send_kwargs)   File "/home/airflow/.local/lib/python3.7/site-packages/requests/sessions.py", line 655, in send     r = adapter.send(request, **kwargs)   File "/home/airflow/.local/lib/python3.7/site-packages/requests/adapters.py", line 516, in send     raise ConnectionError(e, request=request) requests.exceptions.ConnectionError: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /5989ece0-f90e-40bf-9c79-1a7beccdb861/oauth2/v2.0/token (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f7d5ed74490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

### Operating System

PRETTY_NAME="Debian GNU/Linux 10 (buster)" NAME="Debian GNU/Linux" VERSION_ID="10" VERSION="10 (buster)" VERSION_CODENAME=buster ID=debian HOME_URL="https://www.debian.org/" SUPPORT_URL="https://www.debian.org/support" BUG_REPORT_URL="https://bugs.debian.org/"

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by tirkarthi** [comment]
> requests.exceptions.ConnectionError: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): 
Max retries exceeded with url: /5989ece0-f90e-40bf-9c79-1a7beccdb861/oauth2/v2.0/token (Caused by
NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f7d5ed74490>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

This looks more like a network issue in accessing 'login.microsoftonline.com' than an Airflow related issue.

**Reply by potiuk** [reply]
> This looks more like a network issue in accessing 'login.microsoftonline.com' than an Airflow related issue.

Definitiely. @seyal84  - please pay attention to the communication when you open an issue. If you are unsure if this is an Airflow issue and cannot provide reproducibility steps - opening an issue makes very little sense - open discussion instead.
